### PR TITLE
common: add isDialIn participant property

### PR DESCRIPTION
**Pull request by kevinwhereby**
-------

### Description

**Summary:**
We added this property to the signal client, so we should add it here to
make it visible to SDK clients

**Related Issue:**
https://linear.app/whereby/issue/COB-927/add-dial-in-client-property-to-browser-sdk

### Testing
1. if you haven't already, create a twilio account, buy a phone number in a convenient country, and add the account SID and auth token to your local stack config like so:
``` json
    "twilio-account-sid": "<account sid>",
    "twilio-auth-token": "<auth token>",
```
1. you'll also need an ngrok account with the auth token configured in local stack:
``` json
    "ngrok-auth-token": "<auth token>",
```
1. start local stack!
1. build the dial-in image in the recording-service repo with `yarn dial-in:build-for-local-stack`
1. enable the `dialInOn` flag for your local embedded build org
1. create a room in your local stack embedded build org with dial-in enabled like so (make sure you include the dialIn field so you get the dial in info in the response!):
```json
{
    "endDate": "2024-12-06T11:54:05.323Z",
    "isLocked": true,
    "roomMode": "normal",
    "roomNamePrefix": "dial-in-",
    "roomNamePattern": "human-short",
    "isDialInEnabled": true,
    "fields": ["dialIn"]
}
```
make note of the returned dial-in code, we'll use this to join the room from a phone:
``` json
{
      "startDate": "2024-07-31T15:30:02.766Z",
      "endDate": "2024-12-06T11:54:05.323Z",
      "roomName": "/dial-in-t7b4tf",
      "roomUrl": "https://kevin-embedded.whereby.com/dial-in-t7b4tf",
      "meetingId": "89071433",
      "dialIn": {
        "code": "5747754832",          <- Here
        "phoneNumbers": [
          {
            "countryCode": "US",
            "phoneNumber": "+1 (415) 488-8450"
          },
          {
            "countryCode": "GB",
            "phoneNumber": "+44 1433 449947"
          }
        ]
      }
    }
```
1. build this SDK branch with local-stack API and Signal addresses
1. Add your room URL to the telehealth embeded test app and `yarn dev:telehealth-app` to start it up
1. add a silly console.log message in the teleheath app's App.tsx line ~24 so we can see the list of remote participants:
``` typescript
    const { localParticipant, remoteParticipants, screenshares, chatMessages } = state;
    console.log("Remote participants", remoteParticipants);
```
1. join the room from the telehealth app in the browser
1. dial your twilio phone number, and when prompted enter the code above
1. after a few seconds you should join the room as a dial-in participant
1. check the logs of the telehealth app, you should see the remote participant has `isDialIn: true`
``` typescript
{
    "avatarUrl": "",
    "breakoutGroup": "",
    "coLocationGroup": "",
    "deviceCapabilities": {
        "canScreenshare": true
    },
    "deviceId": "9fbc5ec3-81bb-4c2f-b75a-bf2b6af84e0a",
    "displayName": "+447494875798",
    "fromDevice": null,
    "fromMethod": null,
    "id": "9b72ed9c-d665-4df5-b5a7-371038319803",
    "isAudioEnabled": true,
    "isAudioOnlyModeEnabled": false,
    "isCoLocated": false,
    "isDevicePermissionDenied": false,
    "isDialIn": true,                                <- Hurray
    "isOwner": false,
    "isRecording": false,
    "isScreenshareAudioEnabled": false,
    "isVideoEnabled": false,
    "name": "Anonymous 9b72ed9c-d665-4df5-b5a7-371038319803",
    "stickyReaction": null,
    "socketId": "ADdooaqGFJYmb9YIAhEp",
    "isPendingToLeave": false,
    "startedCloudRecordingAt": null,
    "userId": null,
    "externalMetadata": null,
    "externalId": null,
    "stream": null,
    "streams": [
        {
            "id": "0",
            "state": "new_accept"
        }
    ],
    "isLocalParticipant": false,
    "roleName": "host",
    "presentationStream": null,
    "newJoiner": true
}
```

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [x] I have written unit tests (if applicable).
-   [ ] I have updated the documentation (if applicable).
-   [ ] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

File: .changeset/short-tools-worry.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@whereby.com/media": patch
+"@whereby.com/core": patch
```
Comment (nandito) [inline .changeset/short-tools-worry.md:3]: I think this is `minor`.

> Given a version number MAJOR.MINOR.PATCH, increment the:
> MAJOR version when you make incompatible API changes
> MINOR version when you add functionality in a backward compatible manner
> PATCH version when you make backward compatible bug fixes

source: https://semver.org

**Review by nandito — APPROVED** [review]
works as expected 👍 

I added a suggestion regarding the versioning.

File: .changeset/short-tools-worry.md (hunk 1 above)
Comment (thyal) [reply .changeset/short-tools-worry.md:3]: agreed

**Review by thyal — APPROVED** [review]
Tested the whole flow and it works nicely 🚀

File: .changeset/short-tools-worry.md (hunk 1 above)
Comment (kevinwhereby) [reply .changeset/short-tools-worry.md:3]: Ahh will fix

**Pull request merged**